src/ucp/api/ucp.h

```diff
@@ -1996,6 +2022,30 @@ ucs_status_ptr_t ucp_ep_flush_nb(ucp_ep_h ep, unsigned flags,
                                  ucp_send_callback_t cb);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ *
+ * @brief Non-blocking flush of outstanding AMO and RMA operations on the
+ * @ref ucp_ep_h "endpoint".
+ *
+ * This routine flushes all outstanding AMO and RMA communications on the
+ * @ref ucp_ep_h "endpoint". All the AMO and RMA operations issued on the
+ * @a ep prior to this call are completed both at the origin and at the target
+ * @ref ucp_ep_h "endpoint" when this call returns.
+ *
+ * @param [in] ep        UCP endpoint.
+ * @param [in] param     Operation parameters, see @ref ucp_request_param_t.
+ *
+ * @return NULL                 - The flush operation was completed immediately.
+ * @return UCS_PTR_IS_ERR(_ptr) - The flush operation failed.
+ * @return otherwise            - Flush operation was scheduled and can be
+ *                                completed in any point in time. The request
+ *                                handle is returned to the application in
+ *                                order to track progress.
+ */
+ucs_status_ptr_t ucp_ep_flush_nbx(ucp_ep_h ep, ucp_request_param_t *param);
```

Comment:
why removed flush flags?
same for ucp_worker_flush_nbx

Reply:
flags value should be moved into params->op_flags